### PR TITLE
fix(rust): use the transaction api properly with sqlx

### PR DESCRIPTION
**Pull request by etorreborre**
This PR fixes the use of transactions with `sqlx`

### CODE REVIEW

File: implementations/rust/ockam/ockam_identity/src/identities/storage/identity_attributes_repository_sql.rs [hunk 1]
```diff
@@ -74,17 +74,13 @@ impl IdentityAttributesRepository for IdentityAttributesSqlxDatabase {
         attribute_name: Vec<u8>,
         attribute_value: Vec<u8>,
     ) -> Result<()> {
-        let transaction = self.database.begin().await.into_core()?;
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_identity/src/identities/storage/identity_attributes_repository_sql.rs:77]: Why do you think transaction is not required here?

File: implementations/rust/ockam/ockam_api/src/cli_state/storage/identities_repository_sql.rs [hunk 2]
```diff
@@ -40,7 +40,6 @@ impl IdentitiesRepository for IdentitiesSqlxDatabase {
         name: &str,
         vault_name: &str,
     ) -> Result<NamedIdentity> {
-        let transaction = self.database.begin().await.into_core()?;
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_api/src/cli_state/storage/identities_repository_sql.rs:43]: Why do you think transaction is not required here?

File: implementations/rust/ockam/ockam_api/src/cli_state/storage/projects_repository_sql.rs [hunk 3]
```diff
@@ -49,14 +49,14 @@ impl ProjectsSqlxDatabase {
 #[async_trait]
 impl ProjectsRepository for ProjectsSqlxDatabase {
     async fn store_project(&self, project: &Project) -> Result<()> {
-        let transaction = self.database.begin().await.into_core()?;
```
Comment (SanjoDeundiak) [inline implementations/rust/ockam/ockam_api/src/cli_state/storage/projects_repository_sql.rs:52]: Why do you think transaction is not required here?

**Review by SanjoDeundiak — CHANGES_REQUESTED** [review]

File: implementations/rust/ockam/ockam_api/src/cli_state/storage/identities_repository_sql.rs (hunk 2 above)
Comment (etorreborre) [reply implementations/rust/ockam/ockam_api/src/cli_state/storage/identities_repository_sql.rs:43]: In the 3 examples you pointed out, the code is reusing an existing query which is not executed with the transaction that was just opened. Some better code could be to include the query (not its execution) inside the transaction. But maybe some better behaviour would be to forbid an update for a `NamedIdentity` altogether. 
I think it's worth spending some time improving the storage semantics of those named/default entities but I propose that we do it in another PR.

**Review by SanjoDeundiak — APPROVED** [review]

**Pull request merged**